core/evancore.vue.libs/src/components/dapp-wrapper/dapp-wrapper.ts

```diff
@@ -502,43 +540,45 @@ export default class DAppWrapperComponent extends mixins(EvanComponent) {
   /**
    * Load the mail information for the current user
    */
-  async loadMails(mailsToReach = 5) {
+  async loadMails(): Promise<void> {
     if (!this.userInfo.mailsLoading) {
       this.userInfo.mailsLoading = true;
 
-      const runtime = this.$store.state.runtime;
+      const { runtime } = this.$store.state;
       try {
         // load mail inbox informations, load 10 for checking for +9 new mails
         this.userInfo.readMails = JSON.parse(window.localStorage['evan-mail-read'] || '[ ]');
 
-        let mails = [ ];
+        let mails = [];
         let offset = 0;
         let initial = true;
         this.userInfo.totalMails = 0;
 
         // load until 5 mails could be decrypted or the maximum amount of mails is reached
         while (mails.length < 5 && (initial || offset < this.userInfo.totalMails)) {
+          // eslint-disable-next-line no-await-in-loop
           const mailResult = await runtime.mailbox.getReceivedMails(5, offset);
```

Comment:
This could actually lead to weird errors, since the loop will continue while it waits for the result.
May push it in an array of Promises and then wait for all promises or use a `for .. of` loop.

Reply:
This is intended. I only want to search mails, until the logic could encrypt 5 mails. Could be the first run or the fifths run, we never know how many loop we need.

Reply:
did you really tested it? IMO it could/would trigger the loop instantly a few times before even the first is resolved.

Reply:
Hmmmmm... Was also unsure, create a short test. It should work:

```js
asyncFunc = async () => await new Promise(resolve => setTimeout(resolve, 1000))
startDate = Date.now()
index = 0

while (index < 10) {
  await asyncFunc();
  console.log(`${(Date.now() - startDate)/1000}s - ${index}`)

  index++
}
```

1.147s - 0
VM593:7 4.054s - 1
VM593:7 5.58s - 2
VM593:7 7.818s - 3
VM593:7 9.724s - 4
VM593:7 11.772s - 5
VM593:7 13.759s - 6
VM593:7 15.761s - 7
VM593:7 17.774s - 8
VM593:7 19.767s - 9